### ISSUE TITLE
Code formatter, organize imports and code cleanup

### ISSUE THREAD

**Issue by pauldaustin** [maven-nar/nar-maven-plugin#167]
I'm not sure what IDE people use to develop the Nar plugin (I use eclipse).

On other projects I have an eclipse configuration for formatting and code cleanup. I then have eclipse to run the code cleanup before save. This means that methods are always ordered in alphabetical order, formatted according to the style guidelines and unused imports are removed or .* imports are converted to full imports.

By doing this it reduces the diffs on commits as it makes sure the formatting is consistent.

I also add a .gitattributes file to define the standard line endings (UNIX) for most files (*.java, *.xml) except windows specific files (e.g. *.bat, *.cmd) and binary (*.exe, *.png).

If this is something the team is interested in I can add sample config and instructions to a branch and create a pull request. Then you can review them.

FYI for the code formatting I basically follow the Sun Java Code Conventions http://www.oracle.com/technetwork/java/codeconvtoc-136057.html.

2 spaces for indent and no tabs. I've found 2 spaces is readable while compact compared with 4 spaces. Also placing the { on the same line for compactness.

**Comment by dscho** [comment]
> On other projects I have an eclipse configuration for formatting and code cleanup.

NAR has that, too: https://github.com/maven-nar/nar-maven-plugin/tree/master/config

**Comment by pauldaustin** [comment]
Great I have added that config to my eclipse to use for the project.

Could you add a Java\Code Style\Clean Up configuration too?

And a .gitattributes? See sample below

````
* text=auto

# text lf
*.c text eol=lf
*.css text eol=lf
*.cxx text eol=lf
*.h text eol=lf
*.i text eol=lf
*.ini text eol=crlf
*.java text eol=lf
*.jsp text eol=lf
*.properties text eol=lf
*.py text eol=lf
*.txt text eol=lf
*.xml text eol=lf
*.xsd text eol=lf
*.rglayer text eol=lf
*.rgobject text eol=lf

#text crlf
*.bat text eol=crlf
*.cmd text eol=crlf
*.ini text eol=crlf
*.exe.manifest text eol=crlf

#binary
*.exe binary
*.ftp binary
*.png binary
*.jpg binary
*.exe binary
*.ddl binary
*.exp binary
*.lib binary
*.jar binary
*.zip binary
*.dylib binary
*.pdb binary
````

**Comment by ctrueden** [comment]
A PR adding that stuff, as well as maybe applying the styles across the codebase, would be very welcome!

**Comment by pauldaustin** [comment]
I'll get working on a PR

**Comment by ctrueden** [comment]
Resolved with the merge of #168. Thanks @pauldaustin!
